syntax = "proto3";

package library;

// 도서관 검색 서비스
service LibraryService {
  // 도서관 소장 여부 확인
  rpc CheckLibrary(LibraryRequest) returns (LibraryResponse);

  // 건강 체크
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

// 요청 메시지
message LibraryRequest {
  string isbn = 1;      // ISBN (선택)
  string title = 2;     // 제목 (선택)
}

// 응답 메시지
message LibraryResponse {
  bool found = 1;              // 소장 여부
  bool available = 2;          // 대출 가능 여부
  string location = 3;         // 위치 (예: 중앙도서관)
  string call_number = 4;      // 청구기호
  string detail_url = 5;       // 상세 URL
  string error_message = 6;    // 에러 메시지
}

// 건강 체크 요청
message HealthCheckRequest {}

// 건강 체크 응답
message HealthCheckResponse {
  string status = 1;
  string version = 2;
}

